index.js

```diff
@@ -167,19 +182,39 @@ module.exports = {
       }
     }), 'whatwg-fetch');
 
-    const polyfillNode = debug(concat(new MergeTrees([abortcontrollerNode, fetchNode]), {
-      inputFiles: ['abortcontroller.js', 'fetch.js'],
+    let inputNodes = [abortcontrollerNode, fetchNode];
+    let inputFiles = ['abortcontroller.js', 'fetch.js'];
+
+    if (!needsFetchPolyfill && needsAbortControllerPolyfill) {
+      inputNodes = [abortcontrollerNode];
+      inputFiles = ['abortcontroller.js'];
+    }
+
+    const polyfillNode = debug(concat(new MergeTrees(inputNodes), {
+      inputFiles,
       outputFile: 'ember-fetch.js',
       sourceMapConfig: { enabled: false }
     }), 'after-concat');
 
     return debug(new Template(polyfillNode, TEMPLATE_PATH, function(content) {
       return {
-        moduleBody: content
+        moduleBody: needsFetchPolyfill || needsAbortControllerPolyfill ? content : ''
```

Comment:
Not sure if I understand correctly, but it doesn't work with passing an empty array of nodes, you need to pass at least one node.

Reply:
You can use [`allowNon`](https://github.com/broccolijs/broccoli-concat#advanced-usage), it will [allow the content set to empty string](https://github.com/broccolijs/broccoli-concat/blob/master/concat.js#L150).

The current way of checking these 2 booleans is producing wrong result, see my comment on first test result.